src/plugins/hiddenColumns/hiddenColumns.js

```diff
@@ -436,105 +391,26 @@ class HiddenColumns extends BasePlugin {
    * @param {HTMLElement} TH Header's TH element.
    */
   onAfterGetColHeader(column, TH) {
-    if (this.isHidden(column)) {
+    if (!this.settings.indicators || column === -1) {
       return;
     }
 
-    let firstSectionHidden = true;
-    let i = column - 1;
-
-    do {
-      if (!this.isHidden(i)) {
-        firstSectionHidden = false;
-        break;
-      }
-      i -= 1;
-    } while (i >= 0);
+    const physicalColumn = this.hot.toPhysicalColumn(column);
+    const sequenceOfPhysicalIndexes = this.hot.columnIndexMapper.getIndexesSequence();
+    const currentPositionInSourceIndexes = sequenceOfPhysicalIndexes.indexOf(physicalColumn);
+    const classList = [];
 
-    if (firstSectionHidden) {
-      addClass(TH, 'firstVisibleColumn');
+    if (column >= 1 &&
+      this.isHidden(this.hot.toVisualColumn(sequenceOfPhysicalIndexes[currentPositionInSourceIndexes - 1]))) {
```

Comment:
Isn't that `getVisualFromPhysicalIndex`?

```suggestion
      this.isHidden(this.hot.toVisualColumn(toPhysicalColumn[currentPositionInSourceIndexes - 1]))) {
```

Reply:
This code has been refactored within https://github.com/handsontable/handsontable/commit/712d62b6a23582b2ddbd83c4ea720c458e5f85e3. It's so easy now... 😉 I hope you will like it.

Reply:
Love it! So clean ❤️